raiden/tests/integration/rpc/assumptions/test_geth_rpc_assumptions.py

```diff
@@ -26,7 +25,14 @@ def send_transaction():
     # geth keeps the latest 128 blocks before pruning. Unfortunately, this can
     # not be configured to speed this test up.
     non_pruned_blocks = 128
-    while web3.eth.blockNumber < pruned_block_number + non_pruned_blocks + 1:
+    # According to the geth devs if we run a PoA chain (clique, dev mode) and
+    # HEAD-127 doesn't contain any transactions, then the state of HEAD-127 will
+    # be the same as HEAD-128, so it will still be referenced and not deleted.
```

Comment:
I understand this as:

- If the block 127 has a transaction, then the block 128 will be pruned.

and the corollaries:

- If the block 127 does not have a transaction, then the block 128 will not be pruned.
- Transactions on the blocks 126 and younger, do not matter for the pruning of the 128th block.

Reply:
I process the statement by the geth devs in the same way you did.

Reply:
This test is tricky, could you please improve the comments so that someone that didn't read our discussion here can understand the rationale behind the test?